src/quicktype-core/Renderer.ts

```diff
@@ -266,10 +290,20 @@ export abstract class Renderer {
         return assignNames(this.setUpNaming());
     }
 
+    protected initializeEmitContextForFilename(filename: string): void {
+        if (this._finishedEmitContexts.has(filename.toLowerCase().toString())) {
+            const existingEmitContext = this._finishedEmitContexts.get(filename.toLowerCase().toString());
+            if (existingEmitContext !== undefined) {
+                this._emitContext = existingEmitContext;
+            }
+        }
+    }
+
     protected finishFile(filename: string): void {
         assert(!this._finishedFiles.has(filename), `Tried to emit file ${filename} more than once`);
         const source = sourcelikeToSource(this._emitContext.source);
         this._finishedFiles.set(filename, source);
+        this._finishedEmitContexts.set(filename.toLowerCase().toString(), this._emitContext);
```

Comment:
This looks weird.  Why is the filename lowercased, and why is `toString()` called on it?

Reply:
Because I mistakenly thought `toLowerCase()` returned a `String` not a `string`. Fixed.